account_authentications.go

```diff
@@ -120,19 +186,39 @@ func (h AccountAuthenticationsHandler) Create(accountId string, a AccountAuthent
 	return &rsp, nil
 }
 
-// Update updates an account authentication method
-func (h AccountAuthenticationsHandler) Update(accountId string, a AccountAuthenticationMethod) (*AccountAuthenticationResponse, error) {
-	if accountId == "" || a.Id == "" {
+// Update updates an account authentication method empty fields are omitted, acts like PATCH
+func (h AccountAuthenticationsHandler) Update(accountId, accountAuthMethID string, a AccountAuthenticationMethodUpdateRequest) (*AccountAuthenticationMethodUpdateResponse, error) {
+	if accountId == "" || accountAuthMethID == "" {
 		return nil, errors.New("cannot update an account authentication method when account id or auth id is empty")
 	}
 
-	path := buildPath("account", accountId, "authentication", a.Id)
+	path := buildPath("account", accountId, "authentication", accountAuthMethID)
 	bts, err := h.client.doPutRequest(path, a)
 	if err != nil {
 		return nil, err
 	}
 
-	var rsp AccountAuthenticationResponse
+	var rsp AccountAuthenticationMethodUpdateResponse
+	if errR := checkAPIResponse(bts, &rsp); errR != nil {
+		return nil, errR
+	}
+
+	return &rsp, nil
+}
+
+// Replace sets all fields from given DTO to account authentication method, acts like PUT
+func (h AccountAuthenticationsHandler) Replace(accountId, accountAuthMethID string, a AccountAuthenticationMethodReplaceRequest) (*AccountAuthenticationMethodUpdateResponse, error) {
```

Comment:
There is something strange here; we use the same API PUT endpoint in the `Replace` method and the `Update` method. I reckon we shouldn't invent anything artificially, and one function should correspond to the API endpoint. 

Moreover, from the back-end perspective, it doesn't matter whether we pass the empty value for a field or omit it from the request. PUT endpoint will check if all required fields for a particular type of authentication are set and only then will allow to enable it. 

Also, if we consider how things will look from the TF/K8s perspective it is more convenient to keep fields as omit-empty.

Reply:
When you send a struct with omitempty it doesn't send that fields. Therefore there is no way to send empties to drop values.

Reply:
I have dropped it, it doesn't work